ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -175,120 +146,62 @@ S3SourcePath(const std::string &dir, const std::string &name)
     return (dir + "/" + strippedName);
 }
 
-// Return if the file exists. First checks the cache, and then the S3 Bucket.
+// Return if the file exists, via looking into the S3 Bucket. Also, if the file exists, fetch the
+// file size.
 static int
-S3FileExists(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name, bool *fileExists)
+S3FileExistsAndGetSize(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name,
```

Comment:
This API seems a little confusing. We still have a `S3FileExists` after this change, so why not have this as `S3FileGetSize`?

Reply:
I admit I dont like the name "S3FileExistsAndGetSize", I also did not think hard there. S3FileExists calls S3FileExistsAndGetSize() to check if the file exists, there is a different call that calls this function to get the size. It is sort of an internal function to the two filesystem APIs that we support. Please do suggest better naming.

Reply:
I would just have `S3FileGetSize` and return `ENOENT` if it doesn't exist - assuming the S3 API gives us some way to tell what the problem is. (It also might just return the right code, in which case this all gets simpler.)